dipy/reconst/ivim.py

```diff
@@ -0,0 +1,379 @@
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+import numpy as np
+
+from dipy.core.gradients import gradient_table
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.dti import _min_positive_signal, apparent_diffusion_coef
+from dipy.reconst.dti import TensorModel, mean_diffusivity
+from dipy.reconst.multi_voxel import multi_voxel_fit
+from dipy.reconst.vec_val_sum import vec_val_vect
+from dipy.core.sphere import Sphere
+
+from distutils.version import LooseVersion
+import scipy
+
+SCIPY_LESS_0_17 = LooseVersion(scipy.version.short_version) < '0.17'
+
+if not SCIPY_LESS_0_17:
+    least_squares = scipy.optimize.least_squares
+else:
+    leastsq = scipy.optimize.leastsq
+
+
+def ivim_function(params, bvals):
+    """The Intravoxel incoherent motion (IVIM) model function.
+
+    The IVIM model assumes that biological tissue includes a volume fraction
```

Comment:
Please move these explanations into the `IvimModel.__init__` docstring. I think that's where users are more likely to look for them.